framework/db/BaseActiveRecord.php

```diff
@@ -423,6 +423,42 @@ public function hasAttribute($name)
     }
 
     /**
+     * Returns a value indicating whether a property can be read.
+     * This method is overridden as the attributes can be accessed like properties.
+     *
+     * @param string $name the property name
+     * @param boolean $checkVars whether to treat member variables as properties
+     * @param boolean $checkBehaviors whether to treat behaviors' properties as properties of this component
+     * @return boolean whether the property can be written
+     */
+    public function canGetProperty($name, $checkVars = true, $checkBehaviors = true)
+    {
+        if ($this->hasAttribute($name) || isset($this->_related[$name]) || array_key_exists($name, $this->_related)) {
```

Comment:
Would be faster and cleaner to do something like

```php
return $this->hasAttribute($name)
    || isset($this->_related[$name])
    || array_key_exists($name, $this->_related)
    || parent::canGetProperty($name, $checkVars, $checkBehaviors);
```
If any of those is true, then we should return true.